docs/contracts/StMatic.md

```diff
@@ -406,233 +307,109 @@ Requirements:
-
-```sol
-function setOracle(address _oracle)
-```
-
-#### Parameters
+| `_amount` | `uint256` | Amount to withdraw. |
 
-| Name      | Type      | Description                |
-| --------- | --------- | -------------------------- |
-| `_oracle` | `address` | Address of oracle contract |
+---
 
-### setTreasury()
-
-Set treasury contract address to `_treasury`. This contract is used to accumulate the protocol treasury fee
-
-```sol
-function setTreasury(address _treasury)
+### delegate()
+Delegates the MATIC deposited into the StMatic contract equally to the validators 
+```solidity
+function delegate() 
 ```
 
-#### Parameters
-
-| Name        | Type      | Description                                        |
-| ----------- | --------- | -------------------------------------------------- |
-| `_treasury` | `address` | Address of contract which accumulates treasury fee |
-
-### setInsuranceFund()
-
-Set insuranceFund contract address to `_insuranceFund`.
-This contract is used to accumulate the protocol insurance fee
+---
 
-```sol
-function setInsuranceFund(address _insuranceFund)
+### claimTokens()
+Allows users claim their tokens from the validators. This requires that the user has a pending withdrawal request
+```solidity
+function claimTokens(uint256 _tokenId)
 ```
-
 #### Parameters
 
-| Name             | Type      | Description                                         |
-| ---------------- | --------- | --------------------------------------------------- |
-| `_insuranceFund` | `address` | Address of contract which accumulates insurance fee |
-
-### setWithdrawalCredentials()
-
-Set credentials to withdraw ETH on ETH 2.0 side after the phase 2 is launched to `_withdrawalCredentials`
+| Name              | Type     | Description                                                    |
+| ----------------- | -------- | -------------------------------------------------------------- |
+| `_tokenId` | `uint256` | Token ID of a pending withdrawal request. |
+---
 
-```sol
-function setWithdrawalCredentials(bytes32 _withdrawalCredentials)
+### distributeRewards()
+Distributes rewards claimed from validators to their operators
+```solidity
+function distributeRewards()
 ```
+---
 
-:::note
-Note that `setWithdrawalCredentials` discards all unused signing keys as the signatures are invalidated.
-:::
-
-#### Parameters
-
-| Name                     | Type      | Description                                                                                 |
-| ------------------------ | --------- | ------------------------------------------------------------------------------------------- |
-| `_withdrawalCredentials` | `bytes32` | Hash of withdrawal multisignature key as accepted by the deposit_contract.deposit functione |
-
-### withdraw()
-
-Issues withdrawal request. **Not implemented.**
-
-```sol
-function withdraw(uint256 _amount, bytes32 _pubkeyHash)
+### withdrawTotalDelegated()
+Allows a node operator withdraw the amount staked to a validator
```

Comment:
Allows the NodeOperatorRegistry contract to claim the total amount delegated to a validator share.